shellbase-core/src/main/scala/com/sumologic/shellbase/ShellCommand.scala

```diff
@@ -65,11 +65,21 @@ abstract class ShellCommand(val name: String,
         case None => {
           // do NOT block the command from executing
           import scala.concurrent.ExecutionContext.Implicits.global
+          var tsOpt: Option[String] = None
           Future {
-            postCommandToSlack(commandPath, arguments)
+            tsOpt = postCommandToSlack(commandPath, arguments)
           }
+          val timeNow = System.currentTimeMillis()
 
-          execute(cmdLine)
+          val result = execute(cmdLine)
+
+          Future {
+            val timeDuration = (System.currentTimeMillis() - timeNow) / 60000
+            tsOpt.foreach { ts =>
```

Comment:
This posting only happens now if `execute` returns `true`/`false`. I think the behavior we want is to post in the case of `execute` throwing an exception too.  I'd recommend `try-finally` here